src/coreclr/inc/gcinfo.h

```diff
@@ -59,9 +60,18 @@ struct GCInfoToken
     PTR_VOID Info;
     UINT32 Version;
 
-    static UINT32 ReadyToRunVersionToGcInfoVersion(UINT32 readyToRunMajorVersion)
+    static UINT32 ReadyToRunVersionToGcInfoVersion(UINT32 readyToRunMajorVersion, UINT32 readyToRunMinorVersion)
     {
-        // GcInfo version is current from  ReadyToRun version 2.0
+        static_assert(MINIMUM_READYTORUN_MAJOR_VERSION == 9,
+            "when min version moves ahead, revisit the following logic, also check https://github.com/dotnet/runtime/issues/98871");
```

Comment:
@EgorBo - I am adding a static assert here for the min version change. Hopefully it will be enough to trigger the contingent cleanups.